torch/nn/modules/linear.py

```diff
@@ -73,23 +76,70 @@ def __init__(self, in_features, out_features, bias=True):
         super(Linear, self).__init__()
         self.in_features = in_features
         self.out_features = out_features
-        self.weight = Parameter(torch.Tensor(out_features, in_features))
+
+        sizes = []
+        sizes.append(out_features) \
+            if isinstance(out_features, int) else sizes.extend(out_features)
+        sizes.append(in_features) \
+            if isinstance(in_features, int) else sizes.extend(in_features)
+
+        self.in_prod = self._sizes_prod(in_features)
+        self.out_prod = self._sizes_prod(out_features)
+
+        self.weight = Parameter(torch.empty(sizes))
         if bias:
-            self.bias = Parameter(torch.Tensor(out_features))
+            self.bias = Parameter(torch.empty(self.out_features))
         else:
             self.register_parameter('bias', None)
         self.reset_parameters()
 
+    def _sizes_prod(self, features):
+        assert isinstance(features, (int, Iterable)), (
```

Comment:
Don't use an assert here. Assert means developer (us) error, but in this case your checking if the user passed in something bad.